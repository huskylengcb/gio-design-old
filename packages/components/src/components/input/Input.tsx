import * as React from 'react';
import { Input as AntInput } from 'antd';
import { SearchProps } from 'antd/lib/input';
import classnames from 'classnames';
import { omit } from 'lodash';
import 'antd/lib/input/style/index.css';
import './input.less';

export interface InputProps {
  className?: string,
  disabled?: boolean,
  size?: 'large' | 'small'
  [key: string]: any,
  error?: boolean
};

class Input extends React.Component<InputProps> {
  public static Search: React.ClassType<SearchProps, any, any>
  public render() {
    const className = classnames(
      'gio-input',
      this.props.className,
      {
        ['gio-input--error']: this.props.error,
        large: this.props.size === 'large',
        small: this.props.size === 'small'
      }
    )
    return (
      <AntInput
        {...omit(this.props, ['error'])}
        className={className}
      />
    );
  }
}

export default Input