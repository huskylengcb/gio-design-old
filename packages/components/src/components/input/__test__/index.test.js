import React from 'react';
import { render } from 'enzyme';
import Input from '..';
import Icon from 'antd/lib/icon';


const suffix = <Icon type='search' />;

describe('Input', () => {
  it('renders correctly', () => {
    const testUnit = render(
      <Input key='01'/>
    );
    expect(testUnit).toMatchSnapshot();
  });

  it('renders with props correctly', () => {
    const testUnit = render(
      <Input key='01' size='small' error='true'/>
    );
    expect(testUnit).toMatchSnapshot();
  });

  it('renders with suffix correctly', () => {
    const testUnit = render(
      <Input key='01' size='small' error='true' suffix={suffix}/>
    );
    expect(testUnit).toMatchSnapshot();
  });

});
