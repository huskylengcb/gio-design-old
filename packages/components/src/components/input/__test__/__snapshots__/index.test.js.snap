// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-input gio-input",
      "type": "text",
      "value": "",
    },
    "children": Array [],
    "name": "input",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
      "value": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
      "value": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Input renders with props correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-input ant-input-sm gio-input gio-input-small gio-input--error",
      "type": "text",
      "value": "",
    },
    "children": Array [],
    "name": "input",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
      "value": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
      "value": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Input renders with suffix correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "gio-input gio-input-small gio-input--error ant-input-affix-wrapper ant-input-affix-wrapper-sm",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ant-input ant-input-sm",
          "type": "text",
          "value": "",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ant-input-suffix",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-label": "icon: search",
                "class": "anticon anticon-search",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "",
                    "data-icon": "search",
                    "fill": "currentColor",
                    "focusable": "false",
                    "height": "1em",
                    "viewBox": "64 64 896 896",
                    "width": "1em",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "data-icon": undefined,
                    "fill": undefined,
                    "focusable": undefined,
                    "height": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "data-icon": undefined,
                    "fill": undefined,
                    "focusable": undefined,
                    "height": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                },
              ],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-label": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-label": undefined,
                "class": undefined,
              },
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ant-input-suffix",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-label": "icon: search",
              "class": "anticon anticon-search",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "",
                  "data-icon": "search",
                  "fill": "currentColor",
                  "focusable": "false",
                  "height": "1em",
                  "viewBox": "64 64 896 896",
                  "width": "1em",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "data-icon": undefined,
                  "fill": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "data-icon": undefined,
                  "fill": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
              },
            ],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-label": undefined,
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-label": undefined,
              "class": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ant-input ant-input-sm",
            "type": "text",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
